packages/block-library/src/classic/index.native.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { classic as icon } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import edit from './edit';
+import metadata from './block.native.json';
```

Comment:
In fact, you probably could avoid overriding JavaScript file at all when using `blocks.registerBlockType` filter (hook).